DOC: get docs to build on Windows

kalenkovich opened the pull request:
#### Reference issue
Fixes #8501


#### What does this implement/fix?
There are a few problems preventing docs from building on Windows. This PR is intended to fix them. As of now, it solves three out of four problems described in #8501.

larsoner commented on tutorials/evoked/plot_eeg_erp.py:
```diff
@@ -190,7 +193,8 @@
 # for a grand average across subjects (or conditions).
 grand_average = mne.grand_average(all_evokeds)
-mne.write_evokeds('/tmp/tmp-ave.fif', all_evokeds)
+destination = op.join(tempfile.gettempdir(), 'tmp-ave.fif')
+mne.write_evokeds(destination, all_evokeds)
```
Rather than actually write this, let's just comment it out. People can uncomment it if they want.
```suggestion
# And they can be written to disk like any other evoked data, e.g.:
# mne.write_evokeds('tmp-ave.fif', all_evokeds)
```

larsoner commented:
Actually it looks like the fourth problem can be solved here:

https://github.com/mne-tools/mne-python/blob/master/mne/viz/_brain/_scraper.py#L82

We probably do not need the leading `/` there.

And can you add a `[circle front]` to your next commit message to make CircleCI build the front-page examples? One of those has an animation and it would be good to make sure it still works. Alternatively, you can make some change to https://github.com/mne-tools/mne-python/blob/master/tutorials/source-modeling/plot_mne_dspm_source_localization.py. (CircleCI by default only builds changed examples, so you either need to change that one to get it to render, or add `[circle front]` to tell it to build all front-page ones.)

larsoner commented on doc/install/contributing.rst:
```diff
@@ -275,6 +275,14 @@ And for building our documentation::
+      WARNING: dot command 'dot' cannot be run (needed for graphviz output), check the graphviz_dot setting
+
+   you might have to install `graphviz`_ manually.
+   Make sure to agree to add graphviz to path during installation.
```
Or it looks like you could also do:
```
export PATH=~/Miniconda3/Library/bin/graphviz:$PATH
```
or the equivalent for your system/env. Not sure if this is better, though.

kalenkovich replied:
Yes, adding graphviz to path is a much saner solution than a separate install! I'll fix it. 

I'll assume that graphviz has been installed into the conda environment and make use of the `CONDA_PREFIX` variable.

kalenkovich replied:
fixed by 2bc392ec2c1ad601252660de151b012c07c7ec2c

larsoner commented:
@kalenkovich did you try the fix [above](https://github.com/mne-tools/mne-python/pull/8502#issuecomment-724792976) for brain-scraped examples? You should be able to check by making the change then running:
```
PATTERN=mne_dspm make html_dev-pattern
```
It should only run the mne_dspm_solutions example in tutorials/source-modeling, then you can check the RST output.

It's best to be on the latest `master` version of `sphinx-gallery` because this one will ensure the example is run and RST file written even if the example was stale

kalenkovich commented:
> @kalenkovich did you try the fix [above](https://github.com/mne-tools/mne-python/pull/8502#issuecomment-724792976) for brain-scraped examples? You should be able to check by making the change then running:
> 
> ```
> PATTERN=mne_dspm make html_dev-pattern
> ```
> 
> It should only run the mne_dspm_solutions example in tutorials/source-modeling, then you can check the RST output.
> 
> It's best to be on the latest `master` version of `sphinx-gallery` because this one will ensure the example is run and RST file written even if the example was stale

Not yet :( 
Not sure I'll be able to get to it today, sorry about that.

I think removing the forward slash will make unix-based systems to treat the path as relative. So, it will work on Windows but break on other systems. Adding `[circle front]` will test it for us, right?

larsoner commented:
> I think removing the forward slash will make unix-based systems to treat the path as relative. 

Only if the original path is relative. Based on what you see on Windows it might already be absolute. And `[circle front]` will indeed test things for us

larsoner commented:
Okay I looked at the sphinx-gallery source code:

https://github.com/sphinx-gallery/sphinx-gallery/blob/master/sphinx_gallery/scrapers.py#L372-L374

Will push a fix that fixes things on Windows for me

larsoner commented:
@kalenkovich feel free to add an entry to `latest.inc`, otherwise LGTM +1 for merge

kalenkovich commented:
I added a line to `latest.inc` and fixed the code that adds graphviz to the path so that it actually works in git-bash.

Docs now build without errors on my pc. I'll mark this PR as ready for review then.

drammock commented on doc/changes/latest.inc:
```diff
@@ -46,6 +46,8 @@ Enhancements
 
 - Add ``picks`` parameter to :func:`mne.preprocessing.fix_stim_artifact` to specify which channel needs to be fixed, by `Alex Gramfort`_ (:gh:`8482`)
 
+- Further improved documentation building instructions and execution on Windows by `kalenkovich`_, and `Eric Larson`_
```
```suggestion
- Further improved documentation building instructions and execution on Windows (:gh:`8502` by `kalenkovich`_ and `Eric Larson`_)
```

drammock replied:
our convention for changelog entries just changed in #8442

larsoner commented:
Travis is unrelated, thanks @kalenkovich !

agramfort commented:
thx @kalenkovich !
